drivers/unix/os_unix.cpp

```diff
@@ -487,9 +487,11 @@ String OS_Unix::get_executable_path() const {
 	//fix for running from a symlink
 	char buf[256];
 	memset(buf, 0, 256);
-	readlink("/proc/self/exe", buf, sizeof(buf));
+	ssize_t len = readlink("/proc/self/exe", buf, sizeof(buf));
 	String b;
-	b.parse_utf8(buf);
+	if (len > 0) {
+		b.parse_utf8(buf, len);
```

Comment:
`readlink` returns the size of the read buffer. We should use it to make `parse_utf8` safe when buffer is truncated.

Reply:
:+1: